vignettes/how-to-update-released-site.Rmd

```diff
@@ -14,64 +14,90 @@ knitr::opts_chunk$set(
-  `git checkout main -- _pkgdown.yml .github/workflows/pkgdown.yaml`
-
-  Most likely candidates:
-  - `_pkgdown.yml`
-  - `.github/workflows/pkgdown.yaml`
-  - `Config/Needs/website: tidyverse/tidytemplate` in DESCRIPTION
-  - Any fixes to `README.Rmd` / `README.md` or released roxygen comments. Don't
-    forget to `devtools::build_readme()` or `devtools::document()`, in that
-    case.
-  
-  These are the two main Git commands you'll need:
-  ```
-  git checkout REF -- path/to/a/specific/file
-  git cherry-pick SHA
-  ```
-* Monitor your progress locally with `pkgdown::build_site()`.
-* Push this branch, maybe using `usethis::pr_push()`. Don't bother opening
-  a pull request (the branch will still be created and pushed).
-* Go to your package's GHA page, maybe using `usethis::browse_github_actions()`.
-  Select the pkgdown workflow. Click *Run workflow* and select the branch you
-  just pushed. If there's no dropdown menu for this, that means your pkgdown
-  workflow config is not current.
-* Scrutinize your newly published **released** site and make sure things look
-  right.
-* You can let this branch sit around for a while, in case you need to iterate
-  on this or if you'd like to backport more niceties before your next CRAN
-  release. Or you can delete immediately, if you're happy with the released
-  site.
+This vignette shows you how to update the released version of your site to match the dev version of the site, so the first step is to ensure that the dev site looks the way that you want it.
+
+## Process
+
+If you're up to speed with the basic idea and just want some code to follow here it us.
+Otherwise, read more below.
+
+### Setup
+
+First, make sure you're in the main branch, and you have the latest version:
+
+```{r}
+gert::git_branch_checkout("main")
+gert::git_pull()
+```
+
+Next figure out the released version that we're updating:
+
+```{r}
+ver <- desc::desc_get_version()[1, 1:3]
+```
+
+We'll use this to create the branch that you'll work in:
+
+```{r}
+gert::git_branch_create(paste0("pkgdown-v", ver), paste0("v", ver))
+```
+
+### Backport changes
+
+Now you need to backport changes from the dev site into this branch.
+Run this R code to generate the git code to pull changes for the most common locations:
+
+```{r}
+files <- c(
+  # overall site config 
+  "_pkgdown.yml",
+  # the GHA that'll impenet
+  ".github/workflows/pkgdown.yaml",
+  # readme and vignettes
+  "README.md", "vignettes",
+  # logo and favicon
+  "man/figures/", "pkgdown/", 
+  # Author metadata and Config/Needs/Website
+  "DESCRIPTION"
+)
+
+glue::glue("git checkout v{ver} -- {files}")
+```
+
+If you update the description, you'll also need undo the change to the `Version`:
+
+```{r}
+desc::desc_set_version(ver)
+```
+
+Now build the site locally and check that it looks as you expect:
+
+```{r}
+pkgdown::build_site()
+```
+
+### Publish
+
+Now you need to publish the site.
+First push your branch to GitHub:
+
+```{r}
+usethis:::git_push_first()
```

Comment:
We could use `pr_push()`.

Pro: use an exported user-facing function
Con: have to say "don't bother creating an actual PR"

But I also assume you must have decided where you fall on this

Reply:
Yeah, I found it super confusing to then just close the opened tab.